### PR TITLE
Markdown linting: Fix MD001 errors and enable rule

### LINKED ISSUE #5240: Markdown linting: Header levels should only increment by one level at a time (MD001)
# Bug report

## Description

The Markdown linter reports about 276 issues when the [rule 'MD001'](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md001---header-levels-should-only-increment-by-one-level-at-a-time) is enabled.

### Steps to reproduce

1. In the `_checks/styles/style-rules-prod` change `exclude_rule 'MD001'` to `rule 'MD001'`.
2. In your terminal, run `rake test:md`.

## Test your updates

1. Follow "Steps to reproduce" locally to get list of errors to fix.
2. Preview the HTML locally to ensure your changes did not break formatting in the resulting HTML (run `rake:preview`). For more information, refer to the [README](https://github.com/magento/devdocs#to-build-locally).

**Note**-->Do NOT use [special keywords](https://help.github.com/en/articles/closing-issues-using-keywords#about-issue-references) in your commit message.

## Expected result

All the reported issues are fixed.
The `rake test:md` command returns `No issues found`.

## Additional information

- Create working branches from the [`MD001-integration`](https://github.com/magento/devdocs/tree/MD001-integration) branch
- Use the [`MD001-integration`](https://github.com/magento/devdocs/tree/MD001-integration) branch as the base branch whenc reating PRs

### PR DESCRIPTION
**Pull request by jeff-matthews**
This pull request (PR) fixes all MD001 (Header levels should only increment by one level at a time) errors and enables the rule on production.

All the changes in this PR come from PRs that were reviewed, approved, and merged with the `MD001-integration` branch.

Fixes #5240

### CODE REVIEW

File: guides/v2.2/pattern-library/general/accessibilityguideline/accessibilityGuideline.md [hunk 1]
```diff
@@ -6,11 +6,11 @@ The Magento application should be accessible for all people, whatever their hard
 
 ## Visual
 
-#### Be mindful of text length
```
Comment (dshevtsov) [inline guides/v2.2/pattern-library/general/accessibilityguideline/accessibilityGuideline.md]: This doesn't seem like a good use of h3.

Comment (jeff-matthews) [reply guides/v2.2/pattern-library/general/accessibilityguideline/accessibilityGuideline.md]: Yeah, I agree... What do you suggest @dshevtsov? Maybe we can convert the H3s to unordered lists?

Comment (jeff-matthews) [reply guides/v2.2/pattern-library/general/accessibilityguideline/accessibilityGuideline.md]: See 1282a6b

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**